cpu/stm32f4/periph/spi.c

```diff
@@ -102,6 +111,9 @@ int spi_init_master(spi_t dev, spi_conf_t conf, spi_speed_t speed)
             return -2;
     }
 
+    /* initialize lock */
+    mutex_init(&spi_config[dev].lock);
```

Comment:
@gebart, wouldn't the porting of `spi_init`-calls to `board.c` lead to the same deficit?